lib/setup.js

```diff
@@ -0,0 +1,59 @@
+var osenv = require('osenv');
+var fs = require('fs-extra');
+var controller = require('./controller');
+
+var dir = osenv.home() + '/.tessel';
+var filename = 'id_rsa';
+var filepath = dir + '/' + filename;
+
+// Get the Tessel object
+// TODO this way potentially introduces a race condition
+var tessel;
+controller.getTessel(function (err, Tessel) {
+  if(err) {
+    console.log(err);
+  }
+  tessel = Tessel;
+});
+
+function setup (opts) {
+  // If name specified, set hostname
+  if(opts.name) {
+    setHostname(opts._[1])
+  }
+  // Read the public key and copy it to the Tessel
+  fs.readFile(filepath + '.pub', {encoding: 'utf-8'}, function (err, data) {
+    authTessel(data);
+  });
+}
+
+// Put the specified SSH key in Tessel's auth file
+function authTessel (pubKey) {
+  console.log('Authenticating Tessel with public key...');
+
+  var sourceFile = filepath + '.pub';
+  var authFile = '/etc/dropbear/authorized_keys';
+
+  // Copy pubKey into authFile
+  tessel.connection.exec(commands.appendFile(sourceFile, authFile), function (err, streams) {
+    err && console.log(err);
+    streams.stderr && console.log(streams.stderr);
+    if(!err && !streams.stderr) {
+      console.log('Tessel authenticated with public key.');
+    }
+  });
+}
+
+// Sets the hostname of the Tessel
+function setHostname (name) {
```

Comment:
In terms of code organization, I would add this function as a method of a Tessel object (so you can do `tessel.setName()` because it's an action that you would only take on a single Tessel. (see `deploy.js` or `erase.js` for how to do that).

Also, I'm curious what you think about exporting this functionality and adding it to the command line so someone could just change the name of their Tessel without calling `setup`? Something like `tessel christen NAME`. My thinking is that people may want to change the name of the Tessel more often than when they initially set it up. I suppose it doesn't do a whole lot different from `setup` if `setup` doesn't recreate the `.tessel` folder if it already exists.